internal/api/v1/cart/repository/repository.go

```diff
@@ -0,0 +1,41 @@
+package repository
+
+import (
+	"errors"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+type ICartRepository interface {
+	Foo(ctx *gin.Context) (string, error)
+	Bar(ctx *gin.Context, isShouldError string) (string, error)
+}
+
+type cartRepository struct {
+	db *gorm.DB
+}
+
+func NewCartRepository(db *gorm.DB) ICartRepository {
+	return &cartRepository{db: db}
+}
+
+func (c *cartRepository) Foo(ctx *gin.Context) (string, error) {
```

Comment:
gin.Context better di level delivery/controller aja

Reply:
agree, gin.Context hanya untuk layer controller.

layer service & repository pakai context.Context.

thanks for the reminder, will update it.

Reply:
fixed here https://github.com/spacetronot-research-team/cart-service/pull/2/commits/e8e078fdb0be1b66c00e4256ad00243d357dbd7c